onnxruntime/core/framework/execution_frame.cc

```diff
@@ -184,12 +184,33 @@ void IExecutionFrame::Init(const std::vector<int>& feed_mlvalue_idxs, const std:
   // 4. handle feed in values. these can override initializer values so must be last
   for (size_t idx = 0, end = feed_mlvalue_idxs.size(); idx < end; ++idx) {
     int ort_value_idx = feed_mlvalue_idxs[idx];
-    // we are sharing the underline tensor/object for MLValue
+    // we are sharing the underlying tensor/object for MLValue
     all_values_[ort_value_idx] = feeds[idx];
   }
 }
 
-Status IExecutionFrame::GetOutputs(std::vector<OrtValue>& fetches) {
+void IExecutionFrame::UpdateFeeds(const std::vector<int>& feed_mlvalue_idxs, const std::vector<OrtValue>& feeds) {
```

Comment:
>UpdateFeeds [](start = 22, length = 11)

we need to acknowledge that we are doing something relative dangerous here, and we should add some defensive code to prevent misuse of this function. 

Base on the SAA (single assignment assumption) of onnx graph, every mlvalue in the execution frame should only be assigned once and only once.  

As we are overriding some entries in the "all_values_" vector, we can check if this entry has been assigned with some value before. 
We should error out if we hit any double assignment cases.

Reply:
We get to decide what we want to do :) Partial Graph execution is a new scenario hence rules will change but I agree with you rationale in general. In this case, its not as dangerious as your message seems to imply since OrtValues use shared_prt and will self-destruct when count reaches zero. I will add appropiate code and comments to prevent any potential misue.

---
In reply to: [600739659](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r600739659) [](ancestors = 600739659)

Reply:
Sounds like a plan.

---
In reply to: [600927767](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r600927767) [](ancestors = 600927767)